x/marketmap/types/genesis.go

```diff
@@ -1,23 +1,31 @@
 package types
 
+import "fmt"
+
 // NewGenesisState returns an instance of GenesisState.
 func NewGenesisState(
-	tickers map[string]Ticker,
-	paths map[string]Paths,
-	providers map[string]Providers,
+	marketMap MarketMap,
+	lastUpdated int64,
+	params Params,
 ) GenesisState {
 	return GenesisState{
-		MarketMap: MarketMap{
-			Tickers:   tickers,
-			Paths:     paths,
-			Providers: providers,
-		},
+		MarketMap:   marketMap,
+		LastUpdated: lastUpdated,
+		Params:      params,
 	}
 }
 
 // ValidateBasic performs basic validation on the GenesisState.
 func (gs *GenesisState) ValidateBasic() error {
-	return gs.MarketMap.ValidateBasic()
+	if err := gs.MarketMap.ValidateBasic(); err != nil {
+		return err
+	}
+
+	if gs.LastUpdated < 0 {
```

Comment:
If we don't want negatives can't we just use a uint?

Reply:
Yeah.  But the sdk block height type is an `int64` (extremely annoying) so I just went with consistency with that.  I suppose we will never lose anything if we make it a `uint64` tho